hititer.go

```diff
@@ -120,23 +120,25 @@ func (d *indexData) trigramHitIterator(ng ngram, caseSensitive, fileName bool) (
 	}
 
 	iters := make([]hitIterator, 0, len(variants))
-	for _, v := range variants {
-		if fileName {
-			blob := d.fileNameNgrams[v]
+
+	buildIters := func(ngramGetter ngramIndex) error {
+		for _, v := range variants {
+			sec := ngramGetter.Get(v)
+			blob, err := d.readSectionBlob(sec)
+			if err != nil {
+				return err
+			}
 			if len(blob) > 0 {
 				iters = append(iters, newCompressedPostingIterator(blob, v))
 			}
-			continue
 		}
+		return nil
+	}
 
-		sec := d.ngrams.Get(v)
-		blob, err := d.readSectionBlob(sec)
-		if err != nil {
-			return nil, err
-		}
-		if len(blob) > 0 {
-			iters = append(iters, newCompressedPostingIterator(blob, v))
-		}
+	if fileName {
+		buildIters(d.fileNameNgrams)
+	} else {
+		buildIters(d.ngrams)
```

Comment:
minor: I have no idea if this is more efficient, but you can avoid the closure if you do something like

```
var ngrams ngramIndex
if fileName {
  ngrams = d.fileNameNgrams
} else {
  ngrams = d.ngrams
}
...
```

Reply:
I ended up taking a different approach.